cli/demo_teeracle.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# setup:
+# run all on localhost:
+#   integritee-node purge-chain --dev
+#   integritee-node --dev -lruntime=debug
+#   rm light_client_db.bin
+#   export RUST_LOG=integritee_service=info,ita_stf=debug
+#   integritee-service init_shard
+#   integritee-service shielding-key
+#   integritee-service signing-key
+#   integritee-service run
+#
+# then run this script
+
+# usage:
+#  export RUST_LOG_LOG=integritee-cli=info,ita_stf=info
+#  update_exchange_rate.sh -p <NODEPORT> -P <WORKERPORT> -d <DURATION> -i <WORKER_UPDATE_INTERVAL>
+
+while getopts ":p:P:d:i:" opt; do
+    case $opt in
+        p)
+            NPORT=$OPTARG
+            ;;
+        P)
+            RPORT=$OPTARG
+            ;;
+        d)
+            DURATION=$OPTARG
+            ;;
+        i)
+            INTERVAL=$OPTARG
+            ;;
+    esac
+done
+
+# using default port if none given as arguments
+NPORT=${NPORT:-9944}
+RPORT=${RPORT:-2000}
+DURATION=${DURATION:-48}
+INTERVAL=${INTERVAL:-86400}
+
+echo "Using node-port ${NPORT}"
+echo "Using worker-rpc-port ${RPORT}"
+echo "Using worker market data update interval ${INTERVAL}"
+echo "Count the update events during ${DURATION}"
+echo ""
+
+MARKET_DATA_SRC="https://api.coingecko.com/"
+let "MIN_EXPECTED_NUM_OF_EVENTS=$DURATION/$INTERVAL-1"
+echo "MIN_EXPECTED_NUM_OF_EVENTS ${MIN_EXPECTED_NUM_OF_EVENTS}"
+
+CLIENT="./../bin/integritee-cli -p ${NPORT} -P ${RPORT}"
+
+echo "* Query on-chain enclave registry:"
+${CLIENT} list-workers
+echo ""
+
+# this will always take the first MRENCLAVE found in the registry !!
+read MRENCLAVE <<< $($CLIENT list-workers | awk '/  MRENCLAVE: / { print $2; exit }')
+echo "Reading MRENCLAVE from worker list: ${MRENCLAVE}"
+
+[[ -z $MRENCLAVE ]] && { echo "MRENCLAVE is empty. cannot continue" ; exit 1; }
+echo ""
+
+echo "Listen to exchange rate updated events during ${DURATION} seconds. There is no trusted data source"
+${CLIENT} exchange-rate-events ${DURATION}
+echo ""
+
+read NO_EVENTS <<< $($CLIENT exchange-rate-events ${DURATION} | awk '/  EVENTS_COUNT: / { print $2; exit }')
+echo "Got ${NO_EVENTS} exchange rate updates when no trusted data source is in the whitelist"
+echo ""
+
+echo "Add MRENCLAVE as trusted teeracle for ${MARKET_DATA_SRC}"
+${CLIENT} add-whitelist //Alice ${MARKET_DATA_SRC} ${MRENCLAVE}
+echo "MRENCLAVE in Whitelist for ${MARKET_DATA_SRC}"
+echo ""
+
+echo "Listen to exchange rate updated events during ${DURATION} seconds"
```

Comment:
```suggestion
echo "Listen to exchange rate update events for ${DURATION} seconds"
```

Reply:
Why exchange rate update events ? the event is ExchangeRateUpdated

Reply:
Ah, I see. How about `ExchangeRateUpdated` then?